snippet_parser/core_test.py

```diff
@@ -127,18 +127,24 @@ def test_strip_css_selectors(self):
         self.assertIn('Important stuff!', snippets[0])
 
     def test_strip_attributes(self):
-        _, snippets = self._do_extract(
+        _, [_, snippets] = self._do_extract(
             '<p><span class="theclass">Stuff</span>{citation_needed_tmpl}</p>')
         self.assertNotIn('theclass', snippets[0])
 
     def test_extract_section(self):
         self._cfg.extract = 'section'
-        _, snippets = self._do_extract(
+        _, [_, snippets] = self._do_extract(
             '<p>{citation_needed_tmpl}<p><p>Full</p><p>Section</p>',
             '{{cn}}\n\nFull\n\nSection')
         expected = '<div class="%s"><p>Full</p><p>Section</p></div>' % (
             core.SNIPPET_WRAPPER_CLASS)
         self.assertEqual(expected, snippets[0])
 
+    def test_no_wikicode_in_section_titles(self):
+        _, [section, snippets] = self._do_extract(
+            '<p>Irrelevant HTML content{citation_needed_tmpl}</p>',
+            '## Section title ''with Wikicode''\n\nIrrelevant content')
```

Comment:
Oops, sorry, I think the outer quotes have to be double quotes or Python won't parse the string correctly, and we also need a {{ cn }} somewhere in there:

```python
"## Section title ''with Wikicode''\n\nIrrelevant content{{ cn }}"
```

That should fix [the failing test in your PR](https://travis-ci.org/eggpi/citationhunt/builds/631277260), or I'll take a closer look myself if it doesn't. By the way, you can use the `run-tests.sh` script to run all tests and check for yourself that this passes.